ecs-init/engine/engine.go

```diff
@@ -55,6 +56,15 @@ type Engine struct {
 	nvidiaGPUManager         gpu.GPUManager
 }
 
+type TerminalError struct {
```

Comment:
It looks like this struct could be avoided. All you seem to be using is the actual TerminalFailureAgentExitCode

Reply:
This struct is actually used here - https://github.com/aws/amazon-ecs-init/pull/386/files#diff-26cb465239297dd88c43dcd6bbc17b49b0d5981fc87eee70366e3c44e4bfd642R77 to detect that the type of error returned is a `TerminalError` .

Reply:
I see that, what I meant is you can detect if it's a terminal error just based on `TerminalFailureAgentExitCode`, right?

Reply:
I was going to comment the same but the exit code of the struct does get used for making the string of the error, which i think is useful